mempool/src/pool.rs

```diff
@@ -0,0 +1,330 @@
+use std::cmp::Ord;
+use std::cmp::Ordering;
+use std::collections::BTreeSet;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fmt::Debug;
+use std::rc::Rc;
+
+use parity_scale_codec::Encode;
+use thiserror::Error;
+
+use common::chain::transaction::Transaction;
+use common::chain::transaction::TxInput;
+use common::chain::OutPoint;
+use common::primitives::amount::Amount;
+use common::primitives::Id;
+use common::primitives::Idable;
+use common::primitives::H256;
+
+// TODO this willbe defined elsewhere (some of limits.rs file)
+const MAX_BLOCK_SIZE_BYTES: usize = 1_000_000;
+
+const MEMPOOL_MAX_TXS: usize = 1_000_000;
+
+pub trait Mempool<C> {
+    fn create(chain_state: C) -> Self;
```

Comment:
You probably also need ChainConfig as well.

Reply:
Also I don't think you'll get the chainstate, but a handle to something you can use to communicate with it. At best, you're not getting an object by value, but a shared pointer (aka, Arc in rust).

Reply:
1. "ChainState" here is just the name I gave to the interface the mempool wants to communicate with. We can call it something else if we plan for there to be an object worthier of the name "ChainState".
2. Why do I need ChainConfig?

Reply:
You need the ChainConfig, for example, to get the prefix that you got by making it public, which is wrong as I mentioned in another comment.

All chain configuration should be retrieved from there. There's no constants related to it that you should take them as is.

Reply:
I see what you mean about the prefix. However, for the test purposes I can generate a `ChainConfig` object by calling `create_mainnet`, and get the prefix from there.

Reply:
Yes.